src/ray/gcs/actor_state_accessor.h

```diff
@@ -0,0 +1,83 @@
+#ifndef RAY_GCS_ACTOR_STATE_ACCESSOR_H
+#define RAY_GCS_ACTOR_STATE_ACCESSOR_H
+
+#include "ray/common/id.h"
+#include "ray/gcs/callback.h"
+#include "ray/gcs/tables.h"
+
+namespace ray {
+
+namespace gcs {
+
+class RedisGcsClient;
+
+/// \class ActorStateAccessor
+/// ActorStateAccessor class encapsulates the implementation details of
+/// read or write or subscribe of actor's specification(immutable fields which
+/// determined at submission time, and mutable fields which determined at runtime).
+class ActorStateAccessor {
+ public:
+  explicit ActorStateAccessor(RedisGcsClient &client_impl);
+
+  ~ActorStateAccessor() {}
+
+  /// Get actor specification from gcs asynchronously.
+  ///
+  /// \param job_id The ID of the job (driver or app).
+  /// \param actor_id The ID of actor that is looked up in the GCS.
+  /// \param callback Callback that is called after read data done.
+  /// \return Status
+  Status AsyncGet(const JobID &job_id, const ActorID &actor_id,
+                  const MultiItemCallback<ActorTableData> &callback);
+
+  /// Add a actor to gcs asynchronously.
+  ///
+  /// \param job_id The ID of the job (driver or app).
+  /// \param actor_id The ID of actor that is add to the GCS.
+  /// \param data_ptr The actor that is add to the GCS.
+  /// \param callback Callback that is called after the data has been written to the GCS.
+  /// \return Status
+  Status AsyncAdd(const JobID &job_id, const ActorID &actor_id,
+                  std::shared_ptr<ActorTableData> data_ptr,
+                  const StatusCallback &callback);
+
+  /// Update actor runtime state to gcs asynchronously.
+  ///
+  /// \param job_id The ID of the job (driver or app).
+  /// \param actor_id The ID of actor that is update to the GCS.
+  /// \param data_ptr The actor that is update to the GCS.
+  /// \param callback Callback that is called after the data has been written to the GCS.
+  /// \return Status
+  /// TODO(micafan) Modify parameter 3 from ActorTableData to 'ActorRuntimeState'
```

Comment:
This TODO doesn't explain well, people will be confused about what `ActorRuntimeState` is.

Maybe rephrase this in this way: ``Don't expose the whole `ActorTableData` and only allow updating dynamic states.``